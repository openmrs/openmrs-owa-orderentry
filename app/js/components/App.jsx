import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import OrderEntryPage from './orderEntry/OrderEntryPage';

const App = () => (
  <Router>
    <div id="wrap">
      <Switch>
        <Route
          path="/"
          component={OrderEntryPage}
        />
      </Switch>
    </div>
  </Router>
);

export default App
