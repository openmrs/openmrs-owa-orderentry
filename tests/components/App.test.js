import React from 'react';

import App from '../../app/js/components/App';

describe('Test for routes', () => {
  it('run all the routes', () => {
    const wrapper = shallow(<App /> );
    expect(wrapper).toMatchSnapshot()
  });
});
