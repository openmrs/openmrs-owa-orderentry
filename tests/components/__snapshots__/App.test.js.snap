// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for routes run all the routes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        id="wrap"
      >
        <Switch>
          <Route
            component={[Function]}
            path="/"
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            path="/"
          />
        </Switch>,
        "id": "wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Route
            component={[Function]}
            path="/"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          id="wrap"
        >
          <Switch>
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>,
          "id": "wrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Route
              component={[Function]}
              path="/"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
